include/boost/date_time/date_parsing.hpp

```diff
@@ -29,6 +31,16 @@
 namespace boost {
 namespace date_time {
 
+inline unsigned short string_to_ushort(std::string const& s)
+{
+#ifdef BOOST_NO_CXX11
+  return boost::lexical_cast<unsigned short>(s);
+#else
+  // TODO: lexical cast has some side effect on <s>  that makes the std::stoul work here
+  boost::lexical_cast<unsigned short>(s);
```

Comment:
@JeffGarland This line should be removed, but i was curious how it would run in CI.  In my local environment, the tests pass when the ```lexical_cast``` is called on ```s```, but fail when it is not called.  There is some side effect of ```boost::lexical_cast``` on ```s``` that I don't understand in the current date_time uninttests.  In particular this test i have been looking at [here](https://github.com/boostorg/date_time/blob/develop/test/gregorian/testgreg_serialize.cpp) fails without doing this cast.